nautobot/dcim/filters/__init__.py

```diff
@@ -490,10 +492,27 @@ class Meta:
         fields = ["id", "name", "description"]
 
 
+class HardwareFamilyFilterSet(NautobotFilterSet, NameSearchFilterSet):
+    device_types = NaturalKeyOrPKMultipleChoiceFilter(
+        queryset=DeviceType.objects.all(),
+        to_field_name="model",
+        label="Device types (model or ID)",
+    )
+    has_device_types = RelatedMembershipBooleanFilter(
+        field_name="device_types",
+        label="Has device types",
+    )
+
+    class Meta:
+        model = HardwareFamily
+        fields = ["id", "name", "description", "tags"]
+
+
 class DeviceTypeFilterSet(NautobotFilterSet):
     q = SearchFilter(
         filter_predicates={
             "manufacturer__name": "icontains",
+            "hardware_family__name": "icontains",
```

Comment:
Probably want to be sure to add `select_related("hardware_family")` to the queryset for DeviceTypeListView.